File: miraculous-frontend/miraculous-ui/src/dto/album-link-dto/album-link-dto.ts
export class AlbumLinkDto {

  id: number;
  title: string;
  artistId: number;
  artistLogin: string;
  numberOfTracks: number;
  numberOfLikes: number;
  genres: string[];
  albumCoverPictureUrlName: string;
  dateOfRelease: string;
  isFree: boolean;
  price: number;
  inCartOfCurrentVisitor: boolean;
  likedByCurrentVisitor: boolean;


  constructor(id?: number, title?: string, artistId?: number, artistLogin?: string,
              numberOfTracks?: number, numberOfLikes?: number, genres?: string[],
              albumCoverPictureUrl?: string, dateOfRelease?: string, isFree?: boolean, price?:number,
              isInCartOfCurrentVisitor?: boolean,
              isAlbumLikedByCurrentVisitor?: boolean) {
    this.id = id;
    this.title = title;
    this.artistId = artistId;
    this.artistLogin = artistLogin;
    this.numberOfTracks = numberOfTracks;
    this.numberOfLikes = numberOfLikes;
    this.genres = genres;
    this.albumCoverPictureUrlName = albumCoverPictureUrl;
    this.dateOfRelease = dateOfRelease;
    this.isFree = isFree;
    this.price = price;
    this.inCartOfCurrentVisitor = isInCartOfCurrentVisitor;
    this.likedByCurrentVisitor = isInCartOfCurrentVisitor;
  }
}

File: miraculous-backend/src/main/java/com/example/miraculousbackend/dto/AlbumLinkDto.java
package com.example.miraculousbackend.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AlbumLinkDto {

    private Long id;
    private String title;
    private Long artistId;
    private String artistLogin;
    private Long numberOfTracks;
    private Long numberOfLikes;
    private List<String> genres;
    private String albumCoverPictureUrlName;
    private String dateOfRelease;
    private Boolean isFree;
    private Double price;
    private boolean isLikedByCurrentVisitor;
    private boolean isInCartOfCurrentVisitor;
}
